lib/workers/repository/init/inherited.ts

```diff
@@ -99,5 +101,35 @@ export async function mergeInheritedConfig(
       'Removed global config from inherited config.',
     );
   }
-  return mergeChildConfig(config, filteredConfig);
+
+  const decryptedConfig = await decryptConfig(
+    filteredConfig,
+    config.repository,
+  );
+  let returnConfig = decryptedConfig;
+  if (is.nonEmptyArray(returnConfig.extends)) {
+    // Decrypt after resolving in case the preset contains npm authentication instead
+    logger.debug('Resolving presets found in inherited config');
+    returnConfig = await decryptConfig(
+      await resolveConfigPresets(returnConfig, config, config.ignorePresets),
+      config.repository,
+    );
+    logger.trace({ config: returnConfig }, 'Resolved inherited config');
+    res = await validateConfig('inherit', returnConfig);
+    if (res.errors.length) {
+      logger.warn(
+        { errors: res.errors },
+        'Found errors in presets inside the inherited configuration.',
+      );
+      throw new Error(CONFIG_VALIDATION);
+    }
+    if (res.warnings.length) {
+      logger.warn(
+        { warnings: res.warnings },
+        'Found warnings in presets inside the inherited configuration.',
+      );
+    }
+  }
+
+  return mergeChildConfig(config, returnConfig);
```

Comment:
What happens if there are global configs in the presets? 

We are filtering them out above, but not here.